### ISSUE TITLE
[web:a11y] Autocomplete not compatible with screen reader shortcuts

### ISSUE THREAD

**Issue by yjbanov** [flutter/flutter#156371]
Internal issues:

* b/346586723
* b/346601921
* b/346588887

Reproduction:

* [Live demo](https://fwa11y.web.app/#/auto-complete)
* [Source code](https://github.com/flutter/flutter/tree/master/dev/a11y_assessments)

We received a bouquet of issues filed against the `Autocomplete` widget:

- Accessible names missing for auto-complete items
- Auto-complete user flow cannot be completed using screen reader shortcuts
- Auto-complete item list not reachable using a screen reader

All these issues sound too close to each other to address them one by one. Filing a single issue for us to review what's going on with `AutoComplete`. It is likely that we only need one fix to capture them all.
